server2/Arcadia.Assistant/Arcadia.Assistant.ExternalStorages.SharepointOnline/StorageItemJsonConvertor.cs

```diff
@@ -0,0 +1,119 @@
+                [$"{this.FieldsMapper.GetSharepointField(x => x.AllDayEvent)}"] = item.AllDayEvent ? "true" : "false",
+                [$"{this.FieldsMapper.GetSharepointField(x => x.CalendarEventId)}"] = $"\"{item.CalendarEventId}\""
+            };
+
+            if (item.Id != 0)
+            {
+                properties.Add($"{this.FieldsMapper.GetSharepointField(x => x.Id)}", item.Id.ToString());
+            }
+
+            if (listItemType != null)
+            {
+                properties.Add("__metadata", $"{{ \"type\":\"{listItemType}\" }}");
+            }
+
+            return $"{{ {string.Join(',', properties.Keys.Select(k => $"\"{k}\":{properties[k]}"))} }}";
+        }
+
+        public StorageItem? JsonToStorageItem(JsonElement item)
+        {
+            var allDayEvent = this.GetJsonBool(item, this.FieldsMapper.GetSharepointField(x => x.AllDayEvent));
+            return new StorageItem
+            {
+                Id = this.GetJsonInt(item, this.FieldsMapper.GetSharepointField(x => x.Id)),
+                Title = this.GetJsonString(item, this.FieldsMapper.GetSharepointField(x => x.Title)),
+                Description = this.GetJsonString(item, this.FieldsMapper.GetSharepointField(x => x.Description)),
+                StartDate = this.GetJsonDate(item, this.FieldsMapper.GetSharepointField(x => x.StartDate), allDayEvent, DateTime.MinValue),
+                EndDate = this.GetJsonDate(item, this.FieldsMapper.GetSharepointField(x => x.EndDate), allDayEvent, DateTime.UtcNow),
+                Category = this.GetJsonString(item, this.FieldsMapper.GetSharepointField(x => x.Category)),
+                AllDayEvent = allDayEvent,
+                CalendarEventId = this.GetJsonString(item, this.FieldsMapper.GetSharepointField(x => x.CalendarEventId))
+            };
+        }
+
+        #endregion
+
+        #region private members
+
+        private string GetJsonString(JsonElement item, string propName, string defaultValue = "")
+        {
+            return item.TryGetProperty(propName, out var prop) && prop.ValueKind != JsonValueKind.Null ? prop.GetString() : defaultValue;
+        }
+
+        private int GetJsonInt(JsonElement item, string propName, int defaultValue = 0)
+        {
+            return item.TryGetProperty(propName, out var prop) && prop.ValueKind != JsonValueKind.Null && prop.TryGetInt32(out var val) ? val : defaultValue;
+        }
+
+        private bool GetJsonBool(JsonElement item, string propName, bool defaultValue = false)
+        {
+            return item.TryGetProperty(propName, out var prop) && prop.ValueKind != JsonValueKind.Null ? prop.GetBoolean() : defaultValue;
+        }
+
+        private DateTime GetJsonDate(JsonElement item, string propName, bool dateOnly, DateTime defaultValue)
+        {
+            var date = item.TryGetProperty(propName, out var prop) && prop.ValueKind != JsonValueKind.Null && prop.TryGetDateTime(out var val) ? val : defaultValue;
```

Comment:
`var date = this.GetJsonDate(item, propName, defaultValue)`

Reply:
It used 3 times in code below.

Reply:
What did you mean? This method gets date from json element and does some transformations. This concrete line is a full duplicate of line 114 and can be replaced with appropriate method call.

Reply:
Oh... sorry, another comment. Git today do smth strange things. I fixed this place.

Reply:
Its comment for 'CalendarEventIdField' configuration parameter parse, but i cant find those comment.

Reply:
Ok then.

Reply:
Done